### PR TITLE
Remove more deprecated items in Sphinx 6.0

### PR DESCRIPTION
**Pull request by AA-Turner**
Follows on from #10471

### Feature or Bugfix
- Refactoring

A

### CODE REVIEW

File: sphinx/builders/html/__init__.py [hunk 1]
```diff
@@ -1215,18 +1215,14 @@ def setup_resource_paths(app: Sphinx, pagename: str, templatename: str,
     pathto = context.get('pathto')
 
     # favicon_url
-    favicon = context.get('favicon')
+    favicon = context.get('favicon_url')
     if favicon and not isurl(favicon):
         context['favicon_url'] = pathto('_static/' + favicon, resource=True)
```
Comment (tk0miya) [inline sphinx/builders/html/__init__.py]: It's better to call this variable as `favicon_url`.
```suggestion
    favicon_url = context.get('favicon_url')
    if favicon_url and not isurl(favicon_url):
        context['favicon_url'] = pathto('_static/' + favicon_url, resource=True)
```

File: sphinx/builders/html/__init__.py [hunk 2]
```diff
@@ -1215,18 +1215,14 @@ def setup_resource_paths(app: Sphinx, pagename: str, templatename: str,
...
-    else:
-        context['favicon_url'] = favicon
 
     # logo_url
-    logo = context.get('logo')
+    logo = context.get('logo_url')
     if logo and not isurl(logo):
         context['logo_url'] = pathto('_static/' + logo, resource=True)
```
Comment (tk0miya) [inline sphinx/builders/html/__init__.py]: This is also.
```suggestion
    logo_url = context.get('logo_url')
    if logo_url and not isurl(logo_url):
        context['logo_url'] = pathto('_static/' + logo_url, resource=True)
```

File: sphinx/domains/python.py [hunk 3]
```diff
@@ -788,9 +784,6 @@ def get_signature_prefix(self, sig: str) -> List[nodes.Node]:
         if 'classmethod' in self.options:
             prefix.append(nodes.Text('classmethod'))
             prefix.append(addnodes.desc_sig_space())
-        if 'property' in self.options:
```
Comment (tk0miya) [inline sphinx/domains/python.py:791]: I noticed we should emit a warning for this usage to users during 5.x.

File: sphinx/util/compat.py [hunk 4]
```diff
@@ -1,31 +1,12 @@
...
-    if 'sphinx.ext.autosummary' in sys.modules:
-        from sphinx.ext import autosummary
-        if hasattr(autosummary, '_objects'):
-            autosummary._objects['_app'] = app  # type: ignore
-        else:
-            autosummary._app = app  # type: ignore
-
-
 def setup(app: "Sphinx") -> Dict[str, Any]:
-    app.connect('builder-inited', register_application_for_autosummary, priority=100)
```
Comment (tk0miya) [inline sphinx/util/compat.py:27]: It seems all compatiblity helpers are removed from this module. Let's remove whole of this module!

**Review by tk0miya — APPROVED** [review]
LGTM with nits.

**Comment by tk0miya** [comment]
Could you check the flake8 warnings, please?

File: sphinx/domains/python.py (hunk 3 above)
Comment (AA-Turner) [reply sphinx/domains/python.py:791]: Opened #10603 to track, we should add the warning in 5.0.3 or as early as possible otherwise.

A

**Pull request merged**